packages/components/src/components/editor/condition-builder/ConditionGroup.tsx

```diff
@@ -0,0 +1,66 @@
+import { IvyIcons } from '@axonivy/ui-icons';
+import { Condition, type ConditionProps } from './Condition';
+import { useConditionBuilderContext, type LogicalOperator } from './ConditionBuilderContext';
+import { BasicSelect } from '@/components/common/select/select';
+import { Flex } from '@/components/common/flex/flex';
+import { Label } from '@radix-ui/react-label';
+import { Button } from '@/components/common/button/button';
+
+export interface ConditionGroup {
+  conditions: Condition[];
+  logicalOperator: LogicalOperator;
+}
+
+interface ConditionGroupProps extends Pick<ConditionProps, 'groupIndex'> {
+  group: ConditionGroup;
+  groupCount: number;
+}
+
+export const ConditionGroup = ({ group, groupIndex, groupCount }: ConditionGroupProps) => {
+  const { updateLogicalOperator, addCondition, removeConditionGroup, conditionMode, logicalOperatorOptions } = useConditionBuilderContext();
+
+  return (
+    <Flex direction='column' gap={2} className='condition-builder__group'>
+      <Flex
+        direction='column'
+        style={
+          conditionMode === 'nested-condition'
+            ? { border: 'var(--basic-border', borderRadius: 'var(--border-r2)', padding: 'var(--size-2)' }
+            : undefined
+        }
+        gap={2}
+      >
+        {conditionMode === 'nested-condition' && (
+          <Flex direction='row' justifyContent='space-between'>
+            <Label>{`Group ${groupIndex + 1}`}</Label>
+            <Button onClick={() => removeConditionGroup(groupIndex)} icon={IvyIcons.Trash} aria-label='Remove Group' />
+          </Flex>
+        )}
+        {group.conditions.map((condition, conditionIndex) => (
+          <Condition
+            key={conditionIndex}
+            condition={condition}
+            conditionIndex={conditionIndex}
+            groupIndex={groupIndex}
+            conditionsCount={group.conditions.length}
+          />
+        ))}
+        <Button onClick={() => addCondition(groupIndex)} icon={IvyIcons.Plus} aria-label='Add Condition' variant='outline'>
+          Add Condition
+        </Button>
+      </Flex>
+      {groupIndex < groupCount - 1 && conditionMode === 'nested-condition' && (
+        <BasicSelect
+          items={Object.entries(logicalOperatorOptions)
```

Comment:
this options to items mapping is very global, and should be only once IMO, so not on every rerender.
E.g. in a useMemo at the start of the component.

Reply:
Or what do you think of the idea if I simply do the mapping of logicalOperatorOptions and typeOptions directly in the useConditionBuilderContext hook and directly return a corresponding array of label/value objects?